tests/unit/helpers/GroupHelperTest.php

```diff
@@ -0,0 +1,199 @@
+                            ->from('{{questions}}')
+                            ->where('gid=' . $gid)
+                            ->query()
+                            ->readAll();
+
+        $this->assertNotEquals($firstGroupQuestions[0]->question_order, $currentOrder[0]['question_order'], 'The question id is incorrect.');
+        $this->assertNotEquals($firstGroupQuestions[1]->question_order, $currentOrder[1]['question_order'], 'The question id is incorrect.');
+
+        // Test result.
+        $this->assertArrayHasKey('type', $result, 'The returned value is not correct.');
+        $this->assertSame('success', $result['type'], 'The returned value is not correct.');
+    }
+
+    /**
+     * Test question group change.
+     * Change Q03 from group two to group three.
+     */
+    public function testQuestionGroupChange()
+    {
+        $gid = self::$testSurvey->groups[2]->gid;
+
+        // Check the original order.
+        $currentOrder = \Yii::app()->db->createCommand()->select('title')
+                            ->from('{{questions}}')
+                            ->where('gid=' . $gid)
+                            ->order('question_order, title ASC')
+                            ->query()
+                            ->readAll();
+
+        // Checking initial order status Q05 Q06 and number of questions on the group.
+        $this->assertCount(2, $currentOrder, 'The number of questions in the group is not correct.');
+        $this->assertSame('Q05', $currentOrder[0]['title'], 'The question title is incorrect.');
+        $this->assertSame('Q06', $currentOrder[1]['title'], 'The question title is incorrect.');
+
+        // Change question group.
+        $orgdata = $this->getOrgData(self::$testSurvey->groups, self::$testSurvey->questions);
+
+        // Changing Q03 to another group.
+        $qid = \Yii::app()->db->createCommand()->select('qid')
+                            ->from('{{questions}}')
+                            ->where('title = "Q03"')
+                            ->andWhere('sid=' . self::$surveyId)
+                            ->query()
+                            ->readAll()[0]['qid'];
+
+        $orgdata['q' . $qid] = 'g' . self::$testSurvey->groups[2]->gid;
+
+        $groupHelper = new \LimeSurvey\Models\Services\GroupHelper();
+        $result = $groupHelper->reorderGroup(self::$surveyId, $orgdata);
+
+        // Check the current order.
+        $currentOrder = \Yii::app()->db->createCommand()->select('title')
+                            ->from('{{questions}}')
+                            ->where('gid=' . $gid)
+                            ->order('question_order, title ASC')
+                            ->query()
+                            ->readAll();
+
+        $this->assertCount(3, $currentOrder, 'The number of questions in the group is not correct.');
+        $this->assertSame('Q03', $currentOrder[0]['title'], 'The question title is incorrect.');
```

Comment:
Detail in the issue : _The question title is incorrect for 1st question after moving : have {$currentOrder[0]['title']} , wait for Q03_ ?

I don't remind if it was added automatically : waiting value, returned value. If yes : not needed to add it 2 times (but why not)./